solutions/usaco-1085.mdx

```diff
@@ -45,4 +45,56 @@ print(product)
 
 </PySection>
 
+<CPPSection>
+
+```cpp
+
+#include<bits/stdc++.h>
+
+using namespace std;
+int main(){
+	int n;
+	cin >> n;
+	vector<int> cows(n), stalls(n), possible_places(n);
+	for(int i=0; i<n; i++){
+		cin>>cows[i];
+	}
+	
+	for(int i=0; i<n; i++){
+		cin>>stalls[i];
+	}
+	
+	sort(cows.begin(), cows.end());
+	sort(stalls.begin(), stalls.end());
+	
+	//Create a list containing the number of stalls each cow can use.
+	
+	for(int i=0; i<n; i++){
+		for(int j=0; j<n; j++){
+			if(cows[i]<=stalls[j]){
+				possible_places[i] += 1;
+			}
+		}
+	}
+	
+	//Create a list containing the number of stalls each cow can use.
+	long long int product=1;
+	
+	/*
+	 For each cow, find out the number of stalls it can use in a valid permutation
+	 Use the formula provided in the explanation
+	 Multiply the cumulative product by this value
+	*/
```

Comment:
```suggestion
	/*
	 * For each cow, find out the number of stalls it can use in a valid permutation
	 * Use the formula provided in the explanation
	 * Multiply the cumulative product by this value
	 */